src/backend/core/api/serializers.py

```diff
@@ -378,6 +378,10 @@ def validate_file(self, file):
         if extension_mime_type != magic_mime_type:
             self.context["is_unsafe"] = True
 
+        # mime_type fallback
+        if extension_mime_type is None:
```

Comment:
`extension_mime_type` is a confusing name. `content_type` or `mime_type` is less error prone IMO.

Reply:
This was to differentiate between the mimetype inferred from the file extension and the mimetype inferred from the file content. I believe this distinction is important. Please take a look at the code above.

Reply:
Yes but at the end it is this one used by the serializer as a validated data. The one provided by magic is only used to set a `is_unsafe` data and then seems never used.